### DISCUSSION TITLE
Getting shift-clicked item in InventoryCondition

### DISCUSSION THREAD

**Discussion by MrlnHi** [Minestom/Minestom#652] [Help]
When shift clicking an item into my custom inventory, how can I get the ItemStack that is being shift-clicked? InventoryConditionResult#getCursorItem seems to return ItemStack.AIR here.

**Comment by ghost** [comment]
Wouldn't it be ``InventoryConditionResult#getClickedItem``?

**Reply by MrlnHi** [reply]
While that would definitely make sense, it doesn't seem to be the case either

**Comment by MrlnHi** [comment]
I've been able to fix my problem through using both InventoryClickEvent and InventoryPreClickEvent instead of Inventory Conditions now, maybe there should be Inventory#addPreInventoryCondition and Inventory#addInventoryCondition to make both possible, but I don't really know.

**Reply by emortaldev** [reply]
I believe the inventory system is getting a rework

**Reply by TheMode** [reply]
It is, and I am not against getting feedback.
